openshift/e2e-tests-openshift.sh

```diff
@@ -0,0 +1,228 @@
+    local output="$(oc -n "$1" get cm "$2" -oyaml | grep "$3")"
+    if [[ -n "${output}" ]]; then
+      echo -e "\n${output}"
+      return 0
+    fi
+    echo -n "."
+    sleep 1
+  done
+  echo -e "\n\nERROR: timeout waiting for configmap $1/$2 to contain '$3'"
+  return 1
+}
+
+# Loops until duration (car) is exceeded or command (cdr) returns non-zero
+function timeout() {
+  SECONDS=0; TIMEOUT=$1; shift
+  while eval $*; do
+    sleep 5
+    [[ $SECONDS -gt $TIMEOUT ]] && echo "ERROR: Timed out" && return 1
+  done
+  return 0
+}
+
+function install_knative(){
+  header "Installing Knative"
+
+  echo ">> Patching Knative Serving CatalogSource to reference CI produced images"
+  RELEASE_YAML="https://raw.githubusercontent.com/openshift/knative-serving/$SERVING_RELEASE_BRANCH/openshift/release/knative-serving-ci.yaml"
+  sed "s|--filename=.*|--filename=${RELEASE_YAML}|"  openshift/olm/knative-serving.catalogsource.yaml > knative-serving.catalogsource-ci.yaml
+
+  # Install CatalogSources in OLM namespace
+  oc apply -n $OLM_NAMESPACE -f knative-serving.catalogsource-ci.yaml
+  timeout 900 '[[ $(oc get pods -n $OLM_NAMESPACE | grep -c knative) -eq 0 ]]' || return 1
+  wait_until_pods_running $OLM_NAMESPACE
+
+  # Deploy Knative Operators Serving
+  deploy_knative_operator serving
+
+  # Wait for 6 pods to appear first
+  timeout 900 '[[ $(oc get pods -n $SERVING_NAMESPACE --no-headers | wc -l) -lt 6 ]]' || return 1
+  wait_until_pods_running knative-serving || return 1
+
+  #enable_knative_interaction_with_registry
+
+  # Wait for 2 pods to appear first
+  timeout 900 '[[ $(oc get pods -n istio-system --no-headers | wc -l) -lt 2 ]]' || return 1
+  wait_until_service_has_external_ip istio-system istio-ingressgateway || fail_test "Ingress has no external IP"
+
+  wait_until_hostname_resolves $(kubectl get svc -n istio-system istio-ingressgateway -o jsonpath="{.status.loadBalancer.ingress[0].hostname}")
+
+  header "Knative Installed successfully"
+}
+
+function deploy_knative_operator(){
+  local COMPONENT="knative-$1"
+
+  cat <<-EOF | oc apply -f -
+	apiVersion: v1
+	kind: Namespace
+	metadata:
+	  name: ${COMPONENT}
+	EOF
+  if oc get crd operatorgroups.operators.coreos.com >/dev/null 2>&1; then
+    cat <<-EOF | oc apply -f -
+	apiVersion: operators.coreos.com/v1
+	kind: OperatorGroup
+	metadata:
+	  name: ${COMPONENT}
+	  namespace: ${COMPONENT}
+	EOF
+  fi
+  cat <<-EOF | oc apply -f -
+	apiVersion: operators.coreos.com/v1alpha1
+	kind: Subscription
+	metadata:
+	  name: ${COMPONENT}-subscription
+	  generateName: ${COMPONENT}-
+	  namespace: ${COMPONENT}
+	spec:
+	  source: ${COMPONENT}-operator
+	  sourceNamespace: $OLM_NAMESPACE
+	  name: ${COMPONENT}-operator
+	  channel: alpha
+	EOF
+  cat <<-EOF | oc apply -f -
+  apiVersion: serving.knative.dev/v1alpha1
+  kind: Install
+  metadata:
+    name: ${COMPONENT}
+    namespace: ${COMPONENT}
+	EOF
+}
+
+
+function enable_knative_interaction_with_registry() {
```

Comment:
We're not dealing with registry, I think we don't need this?

Reply:
Do we want to push an image with `kn` included ? There is currently a PR underway upstream which exactly does thi.

Reply:
yeah, basically this is needed for `push` operation, we can keep this.